src/crimson/os/seastore/omap_manager.h

```diff
@@ -114,37 +114,43 @@ class OMapManager {
     size_t max_result_size = 128;
 
     /// true denotes behavior like lower_bound, upper_bound otherwise
-    bool inclusive = false;
+    /// range start behavior
+    bool first_inclusive = false;
+    /// range end behavior
+    bool last_inclusive = false;
 
     omap_list_config_t(
       size_t max_result_size,
-      bool inclusive)
+      bool first_inclusive,
+      bool last_inclusive = false)
       : max_result_size(max_result_size),
-	inclusive(inclusive) {}
+	first_inclusive(first_inclusive),
+	last_inclusive(last_inclusive) {}
     omap_list_config_t() {}
     omap_list_config_t(const omap_list_config_t &) = default;
     omap_list_config_t(omap_list_config_t &&) = default;
     omap_list_config_t &operator=(const omap_list_config_t &) = default;
     omap_list_config_t &operator=(omap_list_config_t &&) = default;
 
-    static omap_list_config_t with_max(size_t max) {
-      omap_list_config_t ret{};
-      ret.max_result_size = max;
-      return ret;
+    auto with_max(size_t max) {
+      this->max_result_size = max;
+      return *this;
     }
 
-    static omap_list_config_t with_inclusive(bool inclusive) {
-      omap_list_config_t ret{};
-      ret.inclusive = inclusive;
-      return ret;
+    auto with_inclusive(
+      bool first_inclusive,
+      bool last_inclusive = false) {
```

Comment:
Let's drop the default argument to be explicit that there are two ends to set up the inclusive.